src/binder/bind_node_visitor.cpp

```diff
@@ -440,6 +441,43 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::InsertStatement> node
           }
 
           ins_val->Accept(common::ManagedPointer(this).CastManagedPointerTo<SqlNodeVisitor>());
+
+          // if we have constant values of variable types being inserted into fixed size columns,
+          // we need to resolve that
+          if (ins_val->GetExpressionType() == parser::ExpressionType::VALUE_CONSTANT) {
+            auto const_val = ins_val.CastManagedPointerTo<parser::ConstantValueExpression>();
+            bool is_variable = const_val->GetReturnValueType() == type::TypeId::VARCHAR ||
```

Comment:
Can we stash `ins_col.TypeModifier()` in a variable up here called `max_length`? It'll clean up the logic below a lot rather than the repeated called and make it more clear what that value is.

Reply:
yep